R/pit.R

```diff
@@ -52,105 +52,74 @@
+#' integer predictions.
+#' @inheritParams ae_median_sample
+#' @return A vector with PIT-values. For continuous forecasts, the vector will
+#' correspond to the length of `true_values`. For integer forecasts, a
+#' randomised PIT will be returned of length
+#' `length(true_values) * n_replicates`
+#' @seealso [pit()]
+#' @importFrom stats runif
 #' @examples
-#'
 #' ## continuous predictions
 #' true_values <- rnorm(30, mean = 1:30)
 #' predictions <- replicate(200, rnorm(n = 30, mean = 1:30))
-#' pit(true_values, predictions)
+#' pit <- pit_sample(true_values, predictions)
+#' plot_pit(pit)
 #'
 #' ## integer predictions
 #' true_values <- rpois(100, lambda = 1:100)
 #' predictions <- replicate(5000, rpois(n = 100, lambda = 1:100))
-#' pit(true_values, predictions,  n_replicates = 5)
-#'
+#' pit <- pit_sample(true_values, predictions, n_replicates = 50)
+#' plot_pit(pit)
 #' @export
 #' @references
 #' Sebastian Funk, Anton Camacho, Adam J. Kucharski, Rachel Lowe,
 #' Rosalind M. Eggo, W. John Edmunds (2019) Assessing the performance of
 #' real-time epidemic forecasts: A case study of Ebola in the Western Area
 #' region of Sierra Leone, 2014-15, <doi:10.1371/journal.pcbi.1006785>
+#' @keywords metric
 
-
-pit <- function(true_values,
-                predictions,
-                plot = TRUE,
-                full_output = FALSE,
-                n_replicates = 50,
-                num_bins = NULL,
-                verbose = FALSE) {
-
-
+pit_sample <- function(true_values,
+                       predictions,
+                       n_replicates = 100) {
 
   # error handling--------------------------------------------------------------
   # check al arguments are provided
   if (!all(c(methods::hasArg("true_values"), methods::hasArg("predictions")))) {
-    stop("`true_values` or `predictions` missing in function 'pit()'")
+    stop("`true_values` or `predictions` missing in function 'pit_sample()'")
   }
   check_not_null(true_values = true_values, predictions = predictions)
 
   # check if there is more than one observation
   n <- length(true_values)
   if (n == 1) {
-    if (verbose) {
-      message("you need more than one observation to assess uniformity of the PIT")
-    }
-    out <- list(p_value = NA,
-                sd = NA)
-    if (full_output) {
-      out <- list(p_values = NA,
-                  calibration = NA,
-                  u = NA)
-    }
+    message(
+      "you need more than one observation to assess uniformity of the PIT"
+    )
+    return(NA)
   }
 
   # check and handle format of predictions
   if (is.data.frame(predictions)) {
     predictions <- as.matrix(predictions)
   }
   if (!is.matrix(predictions)) {
-    msg <- sprintf("'predictions' should be a matrix. Instead `%s` was found",
-                   class(predictions[1]))
+    msg <- sprintf(
+      "'predictions' should be a matrix. Instead `%s` was found",
+      class(predictions[1])
```

Comment:
```suggestion
      class(predictions)[1]
```

?

see comment above